idaes/apps/grid_integration/multiperiod/multiperiod.py

```diff
@@ -33,24 +55,85 @@ def __init__(
         process_model_func,
         linking_variable_func,
         periodic_variable_func=None,
+        use_stochastic_build=False,
+        set_days=None,
+        set_years=None,
+        set_scenarios=None,
+        initialization_func=None,
+        unfix_dof_func=None,
+        flowsheet_options={},
+        initialization_options={},
+        unfix_dof_options={},
+        solver=None,
+        outlvl=logging.WARNING,
     ):  # , state_variable_func=None):
+
+        super().__init__()
+
         self.n_time_points = n_time_points
 
         # user provided functions
         self.create_process_model = process_model_func
         self.get_linking_variable_pairs = linking_variable_func
         self.get_periodic_variable_pairs = periodic_variable_func
+        self.initialization_func = initialization_func
+        self.unfix_dof_func = unfix_dof_func
         # self.get_state_variable_pairs = state_variable_func
 
         # populated on 'build_multi_period_model'
-        self._pyomo_model = None
         self._first_active_time = None
 
+        # Create sets
+        if use_stochastic_build:
+            self.set_time = pyo.RangeSet(n_time_points)
+
+            if set_days is not None:
+                self.set_days = pyo.Set(initialize=set_days)
+                self._multiple_days = True
+
+            else:
+                self._multiple_days = False
+
+            if set_years is not None:
+                self.set_years = pyo.Set(initialize=set_years)
+                self._multiyear = True
+
+            else:
+                self._multiyear = False
+
+            if set_scenarios is not None:
+                self.set_scenarios = pyo.Set(initialize=set_scenarios)
+                self._stochastic_model = True
+
+            else:
+                self._stochastic_model = False
+
+            if solver is None:
+                solver = get_solver()
+
+            logging.basicConfig(level=outlvl)
+            _logger = logging.getLogger(__name__)
+
+            # Build the stochastic multiperiod optimization model
+            self.build_stochastic_multiperiod(
+                flowsheet_options,
+                initialization_options,
+                unfix_dof_options,
+                solver,
+            )
+
         # optional initialzation features
         # self.initialization_points = None   #library of possible initial points
         # self.initialize_func = None         #function to perform the initialize
 
-    def build_multi_period_model(self, model_data_kwargs=None):
+    def build_multi_period_model(
```

Comment:
Very minor, but I noticed some functions use `multi_period` in the function name while others include `multiperiod`. I think it would be nice one form was selected for consistency.

Reply:
Thanks for the suggestion! I prefer `multiperiod`, but the old version uses `multi_period`. So, to keep it backwards compatible, I changed everything to `multi_period`.